protocol_tests/connection/test_connection.py

```diff
@@ -4,13 +4,314 @@
+                'response_requested': True
+            },
+            condition=lambda msg: msg.type == ('https://didcomm.org/trust_ping/1.0/ping_response' 
+                                                or 'did:sov:BzCbsNYhMrjHiqZDTUASHg;spec/trust_ping/1.0/ping_response'),
+            timeout=10,
+        )
+
+        # Resgister a new handler for the out-of-band handshake reuse.
+        handler = HandshakeReuseHandler(invite['@id'])
+        conn.route_module(handler)
+
+        # Let's attempt to reuse the invitation here
+        await handler.send_handshake_reuse(conn)
+
+        # Let's test the connection again with a trust ping
+        await conn.send_and_await_reply_async(
+            {
+                '@type': 'https://didcomm.org/trust_ping/1.0/ping',
+                'response_requested': True
+            },
+            condition=lambda msg: msg.type == ('https://didcomm.org/trust_ping/1.0/ping_response' 
+                                                    or 'did:sov:BzCbsNYhMrjHiqZDTUASHg;spec/trust_ping/1.0/ping_response'),
+            timeout=10,
+        )
+        
+    yield 'flow_complete', conn
+
+
+
+@pytest.fixture
+async def oob_receiver(config, backchannel, temporary_channel):
+    """Out of band receiver flow fixture."""
+    generator = _oob_receiver_flow(config, backchannel, temporary_channel)
+    yield generator
+    await generator.aclose()
+
+
+@pytest.mark.asyncio
+@meta(protocol='out-of-band', version='1.0',
+      role='receiver', name='oob-receiver-flow')
+async def test_oob_receiver_flow(oob_receiver):
+    """Test out of band flow as started by the agent under test."""
+    await run(oob_receiver)    
+
+
+async def _oob_sender_flow(config, backchannel, temporary_channel):
+    """Protocol generator for sender role."""
+    # I create an invite and they use my invite
+
+    # Start off with generating some connection information for the suite
+    with temporary_channel() as conn:
+        invite_verkey_b58 = conn.verkey_b58
+        invite_sigkey = conn.sigkey
+
+        # Make an OOB invitation to send to our AUT
+        invite = OutOfBandInvite.make(
+            'test-suite-oob-connection-started-by-agent',
+            'Test the interoperability of an agent',
+            'p2p-messaging',
+            invite_verkey_b58,
+            config['endpoint'],
+            publicDid=conn.did,
+            handshake_protocols=["https://didcomm.org/connections/1.0", "https://didcomm.org/didexchange/1.0"]
+        )
+
+        yield 'invite', conn, invite
+
+        # Have the AUT use our invitation and wait for the next msg
+        with conn.next() as next_request:
+            await backchannel.out_of_band_v1_0_use_invitation(invite.to_url())
+            msg = await wait_for(next_request, 30)
+        
+        # If the agent wants to preform a connection protocol
+        if msg['@type'] == (ConnectionRequest.TYPE or ConnectionRequest.ALT_TYPE):
```

Comment:
We are currently accepting both message prefixes but when we enter the final phase, we will need to accept only messages with https prefixes.  It would be preferable to make it so that only one place in the test suite needs to be updated when that happens.